support/ray_cluster_client.go

```diff
@@ -45,20 +49,32 @@ type RayJobLogsResponse struct {
 	Logs string `json:"logs"`
 }
 
+type RayClusterClientConfig struct {
+	SkipTlsVerification bool
+}
+
 var _ RayClusterClient = (*rayClusterClient)(nil)
 
 type rayClusterClient struct {
-	endpoint url.URL
+	endpoint    url.URL
+	httpClient  *http.Client
+	bearerToken string
 }
 
 type RayClusterClient interface {
 	CreateJob(job *RayJobSetup) (*RayJobResponse, error)
 	GetJobDetails(jobID string) (*RayJobDetailsResponse, error)
 	GetJobLogs(jobID string) (string, error)
+	GetJobs() ([]map[string]interface{}, error)
+	WaitForJobStatus(test Test, jobID string) string
 }
 
-func NewRayClusterClient(dashboardEndpoint url.URL) RayClusterClient {
-	return &rayClusterClient{endpoint: dashboardEndpoint}
+func NewRayClusterClient(dashboardEndpoint url.URL, config RayClusterClientConfig, bearerToken string) RayClusterClient {
```

Comment:
It may be good to adjust client to look like Prometheus client. Such approach provide more flexibility, wrapping all the configuration in Config.
See https://github.com/project-codeflare/codeflare-common/blob/main/support/prometheus_api_client.go

Reply:
@sutaakar I have updated PR accordingly, Please review !